sources/packages/backend/apps/test-db-seeding/src/seed-executors/seed-executor.ts

```diff
@@ -0,0 +1,134 @@
+    // (which are passed as comma separated  parameter with the npm command).
+    if (classesToBeSeeded && classesToBeSeeded.length > 0) {
+      allMethodsToBeSeeded = [];
+      classesToBeSeeded.forEach((classToBeSeeded) => {
+        allMethodsToBeSeeded = allMethodsToBeSeeded.concat(
+          allMethodsWithMetaData.filter(
+            (eachClass) =>
+              eachClass.class.discoveredClass.name === classToBeSeeded,
+          ),
+        );
+      });
+    }
+
+    // Execute Priority order 1 methods after removing skipped classes and
+    // skipped methods.
+    await this.executeSeedMethods(
+      allMethodsToBeSeeded,
+      SeedPriorityOrder.Priority1,
+    );
+
+    // Execute other (priority order : unknown) methods after removing skipped
+    // classes and skipped methods.
+    await this.executeSeedMethods(allMethodsToBeSeeded);
+  }
+
+  /**
+   * Method to get all discovered methods of the discovered classes.
+   */
+  private async getAllDiscoveredClassesAndMethods(): Promise<ClassMethods[]> {
+    // Get discovered classes.
+    const discoveredClasses =
+      await this.discoveryService.providersWithMetaAtKey<ProviderInterface>(
+        DATA_SEED,
+      );
+
+    const allMetaData = [];
+
+    discoveredClasses.forEach((discoveredClass) => {
+      // Get the discovered methods of the discovered classes.
+      const metaMethods =
+        this.discoveryService.classMethodsWithMetaAtKey<MethodInterface>(
+          discoveredClass.discoveredClass,
+          DATA_SEED_METHOD,
+        );
+
+      metaMethods.forEach((metaMethod) =>
+        allMetaData.push({ class: discoveredClass, method: metaMethod }),
+      );
+    });
+
+    return allMetaData;
+  }
+
+  /**
+   * Executes seed methods as per the seed priority order, after
+   * removing skipped classes and methods.
+   * @param allMethodsToBeSeeded method array to be seeded.
+   * @param order order that need to be seeded. If nothing is passed,
+   * it will execute as seed priority "unknow".
+   */
+  private async executeSeedMethods(
+    allMethodsToBeSeeded: ClassMethods[],
+    order = SeedPriorityOrder.Unknown,
+  ): Promise<void> {
+    // Used to limit the number of asynchronous operations
+    // that will start at the same time.
+    const maxPromisesAllowed = os.cpus().length;
+    // Hold all the promises that must be processed.
+    const promises: Promise<undefined>[] = [];
+
+    for (const eachMethod of allMethodsToBeSeeded) {
+      if (
+        eachMethod.class.meta.order === order &&
+        !(eachMethod.class.meta.skip || eachMethod.method.meta.skip)
+      ) {
+        // Calling the decorated methods.
+        promises.push(
+          eachMethod.method.discoveredMethod.handler.bind(
+            eachMethod.method.discoveredMethod.parentClass.instance,
+          )(),
+        );
+        if (promises.length >= maxPromisesAllowed) {
+          // Waits for all be executed.
+          await Promise.allSettled(promises);
```

Comment:
`allSettled` will ensure that all promises are executed and it will not stop if an error happens.
I believe that if some error happens the processes should be aborted, otherwise some data will be missing and DB seeding will be incomplete.

Reply:
I completely agree, it was added as part of my testing and then missed and removed. nice catch 👍